compiler/optimizer/OSRDefAnalysis.cpp

```diff
@@ -1503,7 +1505,7 @@ int32_t TR_OSRLiveRangeAnalysis::fullAnalysis(bool includeParms, bool containsPe
              //In involuntaryOSR mode, there are cases where the first evaluation point of a node is under a treetop  
              //that is not an OSRPoint but another treetop with the same bci is an OSRPoint. Need to call buildOSRLiveRangeInfo
              //in this case for the first evaluation point to make sure the liveness info is bookkept correctly for this bci
-             (_visitedBCI->isSet(node->getByteCodeInfo().getByteCodeIndex()) && comp()->getOSRMode() == TR::involuntaryOSR)) 
+             (node->getByteCodeInfo().getCallerIndex() == callerIndex && _visitedBCI->isSet(node->getByteCodeInfo().getByteCodeIndex()) && comp()->getOSRMode() == TR::involuntaryOSR)) 
```

Comment:
the safety check is good, but should we ever see one of these? Should this be a fatal assert since we could miss an OSR point with a bad BCI?

Reply:
We shouldn't after this change. Will add an assert